app/models/concerns/generate_csv.rb

```diff
@@ -0,0 +1,26 @@
+module GenerateCsv
+  extend ActiveSupport::Concern
+  require 'csv'
+
+  class_methods do
+    def to_csv
+      CSV.generate(col_sep: ';') do |csv|
+        csv << column_names_for_export
+        order(created_at: :desc).each do |item|
```

Comment:
@MartGS  It's better to use `find_each` method (more optimal for long lists and it does not return the full list of objects as result).